### PR TITLE
test: batch check integration tests

### PR DESCRIPTION
**Pull request by justincoh**
## Description

This PR is entirely integration tests for #2039. It reuses all of our existing yaml-defined Check integration tests. For unit tests see #2065 

## References

#2047 was split into two separate PRs for ease of review. This one is all integration tests

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

If you haven't done so yet, we would appreciate it if you could star the [OpenFGA repository](https://github.com/openfga/openfga). :)

### CODE REVIEW

File: tests/batchcheck/batchcheck_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+func testRunAll(t *testing.T, engine string) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	cfg := config.MustDefaultConfig()
+	cfg.Experimentals = append(cfg.Experimentals, "enable-check-optimizations")
+	cfg.Log.Level = "error"
+	cfg.Datastore.Engine = engine
+	cfg.ListUsersDeadline = 0 // no deadline
```
Comment (willvedd) [inline tests/batchcheck/batchcheck_test.go:39]: This necessary?

File: tests/batchcheck/batchcheck.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+						continue
+					}
+
+					testErr := thisResult.GetError()
+					inputErrorCode := testErr.GetInputError().Number()
+					if inputErrorCode != 0 {
+						require.Equal(t, expected.ErrorCode, int(inputErrorCode))
+					} else {
+						require.Equal(t, expected.ErrorCode, int(testErr.GetInternalError().Number()))
+					}
```
Comment (willvedd) [inline tests/batchcheck/batchcheck.go:158]: Not a big deal but if you want to get rid of the `continue` 🤷‍♂️ 

```suggestion
					if expected.ErrorCode != 0 {
					        testErr := thisResult.GetError()
					        inputErrorCode := testErr.GetInputError().Number()
					        if inputErrorCode != 0 {
						        require.Equal(t, expected.ErrorCode, int(inputErrorCode))
					        } else {
						        require.Equal(t, expected.ErrorCode, int(testErr.GetInternalError().Number()))
					        }
					}
					require.Equal(t, expected.Allowed, thisResult.GetAllowed())
					require.Nil(t, thisResult.GetError())

```

**Review by willvedd — COMMENTED** [review]
LGTM just two small things

File: tests/batchcheck/batchcheck_test.go (hunk 1 above)
Comment (justincoh) [reply tests/batchcheck/batchcheck_test.go:39]: nope

File: tests/batchcheck/batchcheck.go (hunk 2 above)
Comment (justincoh) [reply tests/batchcheck/batchcheck.go:158]: I actually don't know the go community's thoughts on `continue` in general, but I'll leave it cuz the increased indentation bothers my dev ocd

**Review by willvedd — APPROVED** [review]

**Pull request merged**